tests/run/given-eta.scala

```diff
@@ -0,0 +1,19 @@
+class C(val x: Int)
+
+trait D
+   type T
+   def trans(other: T): T
+
+def f(x: Int)(given c: C)(y: Int) = x + c.x + y
+def g(x: Int)(given d: D)(y: d.T): d.T = d.trans(y)
+
+@main def Test =
+  given C(1)
+  val x = f
```

Comment:
What happens with eta-expansion if a parameter in the given parameter list depends on a parameter from a previous parameter list ?

Reply:
It usually fails because it can't find the given (unless the same type happens to be implemented by a previous gjven parameter). I added a test.